src/library_syscall.js

```diff
@@ -301,6 +301,13 @@ var SyscallsLibrary = {
       if (info.allocated) {
         _free(info.malloc);
       }
+    } else {
+      // len < info.len, which means a partial munmap (i.e., not of the entire
+      // allocated region). If we did an actual allocation, this would leak, as
+      // all we have is malloc/free and not a real mmap.
+      if (info.allocated) {
+        abort('munmap of part of an mmap-ed region is not supported');
```

Comment:
I think we can just do `if (len != info.len) { abort('munmap of part of an mmap-ed region is not supported'); }` above on line 285 then un-undent the rest of the function.   

I'm not sure where is any need to add the extra check is there?

Reply:
We could also just return `EINVAL` in release builds to save the extra string data maybe?  Although I guess anyone serious about code side should not be using mmap anyway?